tests/ingest_runs/test_validation.py

```diff
@@ -0,0 +1,242 @@
+
+from tests.extract_configs.fixtures import make_template_df
+
+DATA_REVIEW_FILE_TYPES = ["PDA", "PTC", "PTO", "SAM", "S3S", "FTR"]
+
+
+@pytest.fixture
+def template_version(db, tmpdir, settings):
+    """
+    Return a function that uploads content to a version and produces an
+    analysis for the version. The content will conform to a templated
+    file type
+    """
+    def make_template_version(
+        file_type, filename, version=None, file=None, study=None
+    ):
+        """
+        Upload file content, create version analysis so cols are accessible
+        on the file version during inspection
+        """
+        if version:
+            pass
+        elif file:
+            version = file.versions.first()
+        else:
+            file = File(file_type=file_type, name=file_type, study=study)
+            file.save()
+            version = Version(
+                file_name=filename, key=filename, size=100, root_file=file
+            )
+            version.save()
+
+        df = make_template_df(file_type)
+        df_content = df.to_csv(sep="\t", index=False)
+
+        settings.BASE_DIR = os.path.join(tmpdir, "test")
+        version.root_file.file_type = file_type
+        version.root_file.save()
+        version.key.save(filename, ContentFile(df_content))
+
+        analysis = analyze_version(version)
+        analysis.creator = version.creator
+        analysis.save()
+
+        version.refresh_from_db()
+
+        assert os.path.exists(version.key.path)
+        assert file_type in version.valid_types
+
+        return version
+
+    return make_template_version
+
+
+@pytest.fixture
+def data_review_with_files(db, template_version, data_review):
+    """
+    Upload content (that matches available extract configs) to the
+    versions in the data_review
+    """
+    versions = []
+    for ft in DATA_REVIEW_FILE_TYPES:
+        versions.append(
+            template_version(ft, f"{ft}.tsv", study=data_review.study)
+        )
+    data_review.versions.set(versions)
+    data_review.save()
+    data_review.validation_resultset.delete()
+    data_review.refresh_from_db()
+
+    return data_review
+
+
+def test_run_validation_success(db, clients, mocker, data_review_with_files):
+    """
+    End to end test of run_validation task
+    """
+    vr = ValidationRunFactory(data_review=data_review_with_files)
+    validation_run.run_validation(str(vr.pk))
+    vr.refresh_from_db()
+    assert vr.state == State.COMPLETED
+
+
+def test_run_validation_fail(db, clients, mocker, data_review):
+    """
+    Test run_validation on failure
+    """
+    mocker.patch(
+        "creator.ingest_runs.tasks.validation_run.validate_file_versions",
+        side_effect=Exception
+    )
+    vr = ValidationRunFactory(data_review=data_review)
+    with pytest.raises(Exception):
+        validation_run.run_validation(str(vr.pk))
+        vr.refresh_from_db()
+        assert not vr.success
+        assert vr.state == State.FAILED
```

Comment:
These assertions will pass no matter what since they're inside the raises block
```suggestion
with pytest.raises(Exception):
        validation_run.run_validation(str(vr.pk))        
        vr.refresh_from_db()
assert not vr.success
assert vr.state == State.FAILED
```